app/src/main/java/agersant/polaris/PolarisPlaybackService.java

```diff
@@ -116,15 +159,59 @@ public void onReceive(Context context, Intent intent) {
 		autoSaveHandler.postDelayed(autoSaveRunnable, 5000);
 
 		pushSystemNotification();
+
+		mediaSession = new MediaSessionCompat(this, getPackageName());
+		mediaSession.setCallback(new Callback(player));
+        mediaSession.setActive(true);
+
+		updateMediaSessionState(PlaybackStateCompat.STATE_NONE);
+
+		dataUpdateRunnable = () -> {
+			updateMediaSessionState(player.isPlaying() ? PlaybackStateCompat.STATE_PLAYING : PlaybackStateCompat.STATE_PAUSED);
+			dataUpdateHandler.postDelayed(dataUpdateRunnable, 5000);
+		};
+		dataUpdateHandler = new Handler();
+		startDataUpdates();
+
+	}
+
+	private void updateMediaSessionState(int state) {
+		PlaybackStateCompat.Builder builder = new PlaybackStateCompat.Builder();
+		builder.setActions(PlaybackStateCompat.ACTION_PLAY | PlaybackStateCompat.ACTION_PLAY_PAUSE | PlaybackStateCompat.ACTION_PAUSE | PlaybackStateCompat.ACTION_SKIP_TO_NEXT | PlaybackStateCompat.ACTION_SKIP_TO_PREVIOUS);
+		builder.setState(state, (long)player.getCurrentPosition(), 1);
+		mediaSession.setPlaybackState(builder.build());
+
+		MediaMetadataCompat.Builder metadataBuilder = new MediaMetadataCompat.Builder();
+		if(player.getCurrentItem() != null) {
```

Comment:
Would suggest putting the result of getCurrentItem() in a local variable since it gets called a bunch below.